### DISCUSSION TITLE
Custom feeds/sources are completely unavailable to unprivileged users in windows

### DISCUSSION THREAD

**Discussion by aronbardsley** [microsoft/winget-cli#4120] [General]
### Description of the new feature / enhancement

Hi All,

**Summary of Issue**
In its most basic form, it seems like custom feeds/sources are completely unavailable to unprivileged users in windows. They can't add them to winget or use any feeds at a machine level. Is there a way we can add a source as a privileged user or system account to the machine which would allow any user on that machine to install from that source? (without giving every user administrator privileges to add the source....)

We need to run winget install under the user context because certain software can only be installed under the user profile and not machine wide (e.g. Spotify). Thus we need the source to be available for that user. If we didn't have this limitation, we would of course install software machine-wide but some packages simply do not allow it. 

**Attempt 1: Cannot add source as unprivileged user**
Running the following command as an unprivileged user in windows will give the following output.

**Command:** `winget source add -n <NAME> <FEED URL> -t Microsoft.Rest`
**Output:** "This command requires administrator privileges to execute."

After looking at the source code for the SourceFlow workflow under AppInstallerCLICore, it appears that there is a check before a source can be added to make sure that the user is running with certain administrator security tokens. 

https://github.com/microsoft/winget-cli/blob/34fbaabe25df5353d92333dbe2c968a5a2feb316/src/AppInstallerCLICore/Workflows/SourceFlow.cpp#L132

We cannot make every user an administrator, so this option is ruled out 

**Attempt 2: Adding the source as a different privileged user on the same machine**
I've tried to add a feed via an admin user and as system account on the machine, but neither will add the feed in such a way that the feed will be available to the unprivileged user (i.e. winget source list does not show the feed for the user). 

**Other Source Locations**
I was thinking that there must be a way to directly add a source in another location or mechanism for system wide so we could grant the user the ability to install from that location. I did see a group policy option and of course the sources which are hard coded/built into the installer executable, but neither of these routes are possible given the constraints we are working within.

Appreciate any help on this

**Comment by denelon** [comment]
If users are able to run "WinGet" they can install user-based packages like Spotify without elevation from the default "winget" source.

If the intent is to offer a curated subset of packages for users, then a private instance of the REST source could be used to make the subset of packages available to the user. The existing group policy objects can add such a source as a default source for those users.

**Reply by TDelmore** [reply]
The Group Policy option, isn't one we have given our constraints as this is not scalable for our use case.
 
We have 1000's of environments we are managing with our solution and setting up group policies in each of these environments isn't a feasible solution.
 
If Group Policy is the only mechanism by which a user may obtain access to a private instance of the REST source, then I would suggest an alternative solution needs to be made available.

It doesn't seem reasonable that GP is the only mechanism to make available a private source to a standard/unprivileged user.
